packages/data-broker/src/__tests__/services/dhis/pullers/EventsPuller.test.js

```diff
@@ -86,13 +95,107 @@ describe('EventsPuller', () => {
         invocationArgs: expect.objectContaining({ dataElementCodes: ['POP01', 'DIF01_DHIS'] }),
       }));
   });
 
   describe('data building', () => {
+    describe('pulling events', () => {
+      it('can pull events', async () => {
+        const getEventAnalyticsResponse = {
+          headers: [{ name: 'POP01', column: 'Population 1', valueType: 'NUMBER' }],
+          metaData: {
+            items: {
+              POP01: { name: 'Population 1' },
+            },
+            dimensions: { POP01: [] },
+          },
+          rows: [[5], [4]],
+        };
+
+        dhisApi = createMockDhisApi({ getEventAnalyticsResponse });
+        stubGetDhisApi(dhisApi);
+        const dataElementCodes = ['POP01', 'POP02'];
+        const organisationUnitCodes = ['DL_FACILITY_A'];
+
+        const events = await eventsPuller.pull([dhisApi], [DATA_GROUPS.POP01_GROUP], {
+          dataElementCodes,
+          organisationUnitCodes,
+        });
+
+        expect(events).toEqual([{ dataValues: { POP01: 5 } }, { dataValues: { POP01: 4 } }]);
+      });
+
+      it('can pull data for just the requested tracked entity instance', async () => {
+        // Return data for all children of the parent org unit
+        const getEventAnalyticsStub = ({ organisationUnitCodes }) => {
+          if (organisationUnitCodes[0] !== 'DL_FACILITY_A' || organisationUnitCodes.length !== 1) {
+            return {
+              headers: [],
+              metaData: { items: {}, dimensions: [] },
+              rows: [],
+            };
+          }
+
+          return {
+            headers: [
+              { name: 'DIF01_DHIS', column: 'Different 1', valueType: 'NUMBER' },
+              {
+                name: 'tei',
+                column: 'Tracked entity instance',
+                valueType: 'TEXT',
+                meta: true,
+              },
+            ],
+            metaData: {
+              items: {
+                DIF01_DHIS: { name: 'Different 1' },
+              },
+              dimensions: {
+                DIF01_DHIS: [],
+              },
+            },
+            rows: [
+              [7, 'tracked_entity_id_dl_household_1'],
+              [5, 'tracked_entity_id_dl_household_2'],
+            ],
+          };
+        };
+
+        dhisApi = createMockDhisApi({ getEventAnalyticsStub });
+        stubGetDhisApi(dhisApi);
+        const dataElementCodes = ['DIF01'];
+        const organisationUnitCodes = ['DL_HOUSEHOLD_1'];
+        const hierarchy = 'explore';
+
+        const events = await eventsPuller.pull([dhisApi], [DATA_GROUPS.POP01_GROUP], {
+          dataElementCodes,
+          organisationUnitCodes,
+          hierarchy,
+        });
+
+        expect(events).toEqual([
+          {
+            dataValues: { DIF01: 7 },
+            trackedEntityCode: 'DL_HOUSEHOLD_1',
+            trackedEntityId: 'tracked_entity_id_dl_household_1',
+          },
```

Comment:
Is this response supposed to have orgUnitCode as well? Just from the PR description

Reply:
The response depends on what comes from the `DhisApi`. Since we're mocking that to just return the value for `DIF01` and the tracked entity id, that what the response has. But in reality it'll come back with the org unit code as well (which would be the parent org unit's code). 

I'll add it to the test as it'll make it explicit that the tracked entity code and the org unit code don't need to match 👍